pkg/driver/controller_server.go

```diff
@@ -864,6 +865,20 @@ func (driver *Driver) controllerPublishVolume(
 	publishContext[targetNamesKey] = strings.Join(publishInfo.AccessInfo.BlockDeviceAccessInfo.TargetNames, ",")
 	publishContext[targetScopeKey] = requestedTargetScope
 	publishContext[lunIDKey] = strconv.Itoa(int(publishInfo.AccessInfo.BlockDeviceAccessInfo.LunID))
+
+	// Start of population of target array details
+	log.Tracef("\n PUBLISH INFO ::::: %v", publishInfo)
+
+	if publishInfo.AccessInfo.BlockDeviceAccessInfo.SecondaryBackendDetails.PeerArrayDetails != nil {
```

Comment:
Can you change this to ```if publishInfo.AccessInfo.BlockDeviceAccessInfo.SecondaryBackendDetails != nil``` as it will cause panic otherwise.

Reply:
I could'nt add above IF condition with != nil , since the compiler complains like
```
pkg/driver/controller_server.go:870:60: cannot convert nil to type model.SecondaryBackendDetails
```
I have confirmed the code changes with or without `SecondaryBackendDetails` populated via the CSP

Reply:
> I could'nt add above IF condition with != nil , since the compiler complains like
> 
> ```
> pkg/driver/controller_server.go:870:60: cannot convert nil to type model.SecondaryBackendDetails
> ```
> 
> I have confirmed the code changes with or without `SecondaryBackendDetails` populated via the CSP

ah, sorry i thought its a pointer. sounds good then.